core/src/components/menu/menu.tsx

```diff
@@ -477,11 +477,11 @@ export class Menu implements ComponentInterface, MenuI {
       this.animation = undefined;
     }
     // Create new animation
-    this.animation = await menuController._createAnimation(this.type!, this);
+    const animation = (this.animation = await menuController._createAnimation(this.type!, this));
     if (!config.getBoolean('animated', true)) {
-      this.animation.duration(0);
+      animation.duration(0);
     }
-    this.animation.fill('both');
```

Comment:
TypeScript logged an error saying that `this.animation` was potentially undefined (because that's what the type definition says). Using the local variable ensures that `animation` is always defined due to the `_createAnimation` return type you noted.